xontrib/sh.py

```diff
@@ -29,6 +30,13 @@ def onepath(cmd, **kw):
         first_compatible_shell = None
         check_output_all = ''
         for s in _installed_shells:
+            if len(_installed_shells) == 1:   # don't check syntax for a single shell
+                if s.rstrip('.exe').lower() == 'cmd':
```

Comment:
The structure of code:
1. Get available shells
2. Check syntax
3. Run the command

Please add the changes along this structure and avoid copying the same code twice like here and below.

Reply:
not sure what the copying the same code was refering to, is it still part of the updated code?